lib/ae_mdw/db/name.ex

```diff
@@ -37,6 +42,78 @@ defmodule AeMdw.Db.Name do
     auction_end + :aec_governance.name_claim_max_expiration(proto_vsn(auction_end))
   end
 
+  @spec expirations_mutation(Blocks.height()) :: NamesExpirationMutation.t()
+  def expirations_mutation(height) do
+    name_mspec =
+      Ex2ms.fun do
+        Model.expiration(index: {^height, name}) -> name
+      end
+
+    auction_mspec =
+      Ex2ms.fun do
+        {:expiration, {^height, name}, tm} -> {name, tm}
+      end
+
+    expired_names = :mnesia.dirty_select(Model.ActiveNameExpiration, name_mspec)
+    expired_auctions = :mnesia.dirty_select(Model.AuctionExpiration, auction_mspec)
+
+    NamesExpirationMutation.new(height, expired_names, expired_auctions)
+  end
+
+  @spec expire_name(Blocks.height(), Names.plain_name()) :: :ok
+  def expire_name(height, plain_name) do
```

Comment:
It would be good to keep this to enforce consistency:
https://github.com/aeternity/ae_mdw/blob/ee3465822fecf346acc564e36f9e209faf8444f8/lib/ae_mdw/db/sync/name.ex#L215

This protects from expiring a name due to `Model.ActiveNameExpiration` that are not the last one. With this protection also we could safely remove the `ensure_no_exsiting_active_name_expiration/1` in regards to name state transition. It would be missing only the analysis if removing it might affect the endpoints that depend on proper `Model.ActiveNameExpiration` records state.

Reply:
You're right, I've added it back, thanks!